pkg/daemon/config.go

```diff
@@ -92,3 +106,59 @@ func tryToLoadOldConfig(nodeProfilesJson []byte) ([]ptpv1.PtpProfile, bool) {
 
 	return []ptpv1.PtpProfile{*ptpConfig}, true
 }
+
+func (conf *ptp4lConf) populatePtp4lConf(config *string) error {
+	lines := strings.Split(*config, "\n")
+	var currentSection string
+	conf.sections = make(map[string]ptp4lConfSection)
+
+	for _, line := range lines {
+		if strings.HasPrefix(line, "[") {
+			currentSection = line
+			currentLine := strings.Split(line, "]")
+
+			if len(currentLine) < 2 {
+				return errors.New("Section missing closing ']'")
+			}
+
+			currentSection = fmt.Sprintf("%s]", currentLine[0])
+			section := ptp4lConfSection{}
+			section.options = make(map[string]string)
+			conf.sections[currentSection] = section
+		} else if currentSection != "" {
+			split := strings.IndexByte(line, ' ')
+			if split > 0 {
+				section := conf.sections[currentSection]
+				section.options[line[:split]] = line[split:]
+				conf.sections[currentSection] = section
+			}
+		} else {
+			return errors.New("Config option not in section")
+		}
+	}
+	if conf.sections["[global]"].options == nil {
+		section := ptp4lConfSection{}
+		section.options = make(map[string]string)
+		conf.sections["[global]"] = section
+	}
+	return nil
+}
+
+func (conf *ptp4lConf) renderPtp4lConf() {
```

Comment:
please change this function to return the ifaceString and the config.

This way you can remove them from the struct there is no need to save the information